demos/bubble_shear-goal.py

```diff
@@ -0,0 +1,313 @@
+
+            # Combine the metrics from each exported timestep in the subinterval
+            subinterval_metric.intersect(metric)
+        metrics.append(metric)
+
+    # Normalise the metrics in space and time
+    space_time_normalise(metrics, mesh_seq.time_partition, mp)
+
+    complexities = np.zeros(len(mesh_seq))
+    for i, metric in enumerate(metrics):
+        complexities[i] = metric.complexity()
+        mesh_seq[i] = adapt(mesh_seq[i], metric)
+
+    num_dofs = mesh_seq.count_vertices()
+    num_elem = mesh_seq.count_elements()
+    pyrint(f"Fixed point iteration {iteration}:")
+    for i, (complexity, ndofs, nelem) in enumerate(
+        zip(complexities, num_dofs, num_elem)
+    ):
+        pyrint(
+            f"  subinterval {i}, complexity: {complexity:4.0f}"
+            f", dofs: {ndofs:4d}, elements: {nelem:4d}"
+        )
+
+    return True
+
+
+# Finally, we can define the :class:`GoalOrientedMeshSeq` and run the fixed point iteration.
+# For the purposes of this demo, we divide the time interval into 6 subintervals and only
+# run two iterations of the fixed point iteration, which is not enough to reach convergence. ::
+
+# Reduce the cost of the demo during testing
+test = os.environ.get("GOALIE_REGRESSION_TEST") is not None
+n = 50 if not test else 5
+dt = 0.01 if not test else 0.05
+maxiter = 2 if not test else 1  # maximum number of fixed point iterations
+
+num_subintervals = 6
+meshes = [UnitSquareMesh(n, n) for _ in range(num_subintervals)]
+end_time = period / 2
+time_partition = TimePartition(
+    end_time,
+    len(meshes),
+    dt,
+    fields,
+    num_timesteps_per_export=5,
+)
+
+parameters = GoalOrientedMetricParameters({"maxiter": maxiter})
+msq = GoalOrientedMeshSeq(
+    time_partition,
+    meshes,
+    get_function_spaces=get_function_spaces,
+    get_initial_condition=get_initial_condition,
+    get_bcs=get_bcs,
+    get_form=get_form,
+    get_solver=get_solver,
+    get_qoi=get_qoi,
+    qoi_type="end_time",
+    parameters=parameters,
+)
+solutions, indicators = msq.fixed_point_iteration(adaptor)
+
+# Let us plot the intermediate and final solutions, as well as the final adapted meshes. ::
+
+import matplotlib.pyplot as plt
+from firedrake.pyplot import tripcolor, triplot
+
+fig, axes = plt.subplots(2, 3, figsize=(7.5, 5), sharex=True, sharey=True)
+
+for i, ax in enumerate(axes.flatten()):
+    ax.set_aspect("equal")
+    ax.set_xlim(0, 1)
+    ax.set_ylim(0, 1)
+    # ax.tick_params(axis='both', which='major', labelsize=6)
+
+    # plot the solution at the final subinterval timestep
+    time = time_partition.subintervals[i][-1]
+    tripcolor(solutions["c"]["forward"][i][-1], axes=ax, cmap="coolwarm")
+    triplot(msq[i], axes=ax, interior_kw={"linewidth": 0.08})
+    ax.annotate(f"t={time:.2f}", (0.05, 0.05), color="white")
+
+fig.tight_layout(pad=0.3)
+fig.savefig("bubble_shear-goal_final_meshes.jpg", dpi=300, bbox_inches="tight")
+
+# .. figure:: bubble_shear-goal_final_meshes.jpg
```

Comment:
Could you follow the naming pattern and have this file and its plots use `bubble_shear-goal_oriented*`?

Reply:
Fixed in 262c78a51a99714a4794b55259edf6fc5aea3567.